### PR TITLE
Fix in `_reshape_mask`

### PR DESCRIPTION
**Pull request by obadakhalili**
Addresses the issue in #4109.

The issue appears when `old_dim` has n instances of `next_stride` but with a remainder. This remainder will push the position of the mask and might cross the axis boundary.

### CODE REVIEW

File: test/test_ops.py [hunk 1]
```diff
@@ -986,6 +986,25 @@ def test_pad(self):
     helper_test_op([(3,3)], lambda x: torch.nn.functional.pad(x, (0,0,3,4), value=1), lambda x: x.pad(((3,4), None), value=1))
     helper_test_op([(3,3)], lambda x: torch.nn.functional.pad(x, (0,0,0,0), value=1), lambda x: x.pad((None, None), value=1))
 
+  def test_pad_reshape(self):
+    a = Tensor.ones((2,)).reshape((1, 2))
```
Comment (chenyuxyz) [inline test/test_ops.py:990]: why does it need to only test Tensor.ones? wouldn't random tensor with shape `(1, 2)` work?

**Review by chenyuxyz — COMMENTED** [review]
Thanks for looking into this! Can you also add the case in test/unit/test_shapetracker? test_ops is quite end2end for this fix and would be nice to include something more local

File: test/test_ops.py (hunk 1 above)
Comment (obadakhalili) [reply test/test_ops.py:990]: Oh, it would work actually. I just had to stick with `Tensor.ones` at the beginning to reproduce the exact faulty view mentioned in the example and kept using it afterward. I will update it to use randomly generated tensors from `helper_test_op`.

**Comment by chenyuxyz** [comment]
thanks!
